drivers/TSP01/driver.js

```diff
@@ -0,0 +1,265 @@
+				}
+			},
+			'command_basic_usage'       : 'motion_interact_class_basic',
+			'command_class'             : 'COMMAND_CLASS_SENSOR_BINARY',
+			'command_report'            : 'SENSOR_BINARY_REPORT',
+			'command_report_parser'     : function( report, node ){
+				// If we got a resolved timeout setting, then act on it
+				if (this._resetTimer) {
+					let refNo = Math.random().toString()
+					this.refNo = refNo;
+
+					module.exports._debug(`Got report for ${node.device_data.token}/alarm_motion, reset after ${this._resetTimer} milliseconds if refNo == ${refNo}`)
+					setTimeout(this.resetTimerCallback, this._resetTimer, this, refNo, node)
+				}
+				if (report['Sensor Type'] == 'Motion') {
+					// setTimeout(function() {node.}, 6000)
+					return report['Sensor Value (Raw)'].toString('hex') == 'ff';
+				}
+			}
+		},
+		'alarm_tamper': {
+			'command_class'             : 'COMMAND_CLASS_SENSOR_BINARY',
+			'command_report'            : 'SENSOR_BINARY_REPORT',
+			'command_report_parser'     : function( report ){
+				if (report['Sensor Type'] == 'Tamper') {
+					return report['Sensor Value (Raw)'].toString('hex') == 'ff';
+				}
+			},
+			'optional': true
+		},
+		'measure_luminance': {
+			'command_class'             : 'COMMAND_CLASS_SENSOR_MULTILEVEL',
+			'command_report'            : 'SENSOR_MULTILEVEL_REPORT',
+			'command_report_parser'     : function( report ){
+				if (report['Sensor Type'] == 'Luminance (version 1)') return parseInt(report['Sensor Value (Parsed)']) * 10;
+			},
+			'optional': true
+		},
+		'measure_temperature': {
+			'command_class'             : 'COMMAND_CLASS_SENSOR_MULTILEVEL',
+			'command_report'            : 'SENSOR_MULTILEVEL_REPORT',
+			'command_report_parser'     : function( report ){
+				if (report['Sensor Type'] == 'Temperature (version 1)') return (parseInt(report['Sensor Value (Parsed)']) - 32) * 5 / 9;
+			},
+			'optional': true
+		},
+		'measure_battery': {
+			'command_class': 'COMMAND_CLASS_BATTERY',
+			'command_report': 'BATTERY_REPORT',
+			'command_report_parser': report => {
+				if (report['Battery Level'] === "battery low warning") return 1;
+				
+				return report['Battery Level (Raw)'][0];
+			}
+		}
+	},
+	settings: {
+		"basic_set_level": {
+			"index": 2,
+			"size": 1,
+		},
+		"pir_sensitivity": {
+			"index": 3,
+			"size": 1,
+		},
+		"light_sensitivity": {
+			"index": 4,
+			"size": 1,
+		},
+		"operation_mode": {
+			"index": 5,
+			"size": 1,
+		},
+		"multi_sensor_function_switch": {
+			"index": 6,
+			"size": 1,
+		},
+		"pir_redetect_interval_time": {
+			"index": 8,
+			"size": 1,
+		},
+		"turn_off_light_time": {
+			"index": 9,
+			"size": 1,
+		},
+		"battery_report_time": {
+			"index": 10,
+			"size": 1,
+		},
+		"illumination_report_time": {
+			"index": 12,
+			"size": 1,
+		},
+		"temperature_report_time": {
+			"index": 13,
+			"size": 1,
+		},
+	},
+	beforeInit: function(deviceDataToken, callback) {
```

Comment:
kind of curious why you re-wrote a lot of the beforeInit that is already in the z-wave driver, just for the reset timer (that can be done much easier)